android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenDeviceIconManager.java

```diff
@@ -0,0 +1,107 @@
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Date;
+
+public class LockScreenDeviceIconManager {
+
+    private Context context;
+
+    LockScreenDeviceIconManager(Context context) {
+        this.context = context;
+    }
+
+    boolean updateCachedImage(String iconIUrl) {
+        SharedPreferences sharedPref = this.context.getSharedPreferences("sdl", Context.MODE_PRIVATE);
+        String iconParameters = sharedPref.getString(iconIUrl, null);
+        if(iconParameters == null) {
+            return true;
+        } else {
+            JSONObject jsonObject = null;
+            try {
+                jsonObject = new JSONObject(iconParameters);
+                long lastUpdatedTime = 0;
+                lastUpdatedTime = (long) jsonObject.get("lastUpdatedTime");
+                long currentTime = System.currentTimeMillis();
+
+                long timeDifference = currentTime - lastUpdatedTime;
+                long daysBetweenLastUpdate = timeDifference / (1000 * 60 * 60 * 24);
+
+                return daysBetweenLastUpdate >= 30;
+
+            } catch (JSONException e) {
+                e.printStackTrace();
+                return true;
+            }
+        }
+    }
+
+    void saveFileToCache(Bitmap icon, String iconUrl) {
+        File f = new File(this.context.getCacheDir(), iconUrl);
+        try {
+            f.createNewFile();
+            ByteArrayOutputStream bos = new ByteArrayOutputStream();
+            icon.compress(Bitmap.CompressFormat.PNG, 0 /*ignored for PNG*/, bos);
+            byte[] bitmapdata = bos.toByteArray();
+
+            FileOutputStream fos = null;
+            fos = new FileOutputStream(f);
+            fos.write(bitmapdata);
+            fos.flush();
+            fos.close();
+            JSONObject iconParams;
+
+            iconParams = buildDeviceIconParameters(f.getAbsolutePath());
+            writeDeviceIconParametersToSystemPreferences(iconUrl, iconParams);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    Bitmap getFileFromCache(String iconUrl) {
+        SharedPreferences sharedPref = this.context.getSharedPreferences("sdl", Context.MODE_PRIVATE);
+        String iconParameters = sharedPref.getString(iconUrl, null);
+
+        if (iconParameters != null) {
+            JSONObject jsonObject = null;
+            try {
+                jsonObject = new JSONObject("storedUrl");
+                String storedUrl = jsonObject.getString("storedUrl");
+                return BitmapFactory.decodeFile(storedUrl);
+            } catch (JSONException e) {
+                e.printStackTrace();
+                return null;
+            }
+        } else {
+            return null;
+        }
+    }
+
+    private void writeDeviceIconParametersToSystemPreferences(String iconUrl, JSONObject jsonObject) throws JSONException {
+        SharedPreferences sharedPref = this.context.getSharedPreferences("sdl", Context.MODE_PRIVATE);
```

Comment:
I would use a more descriptive name and create a constant for that string instead of hard-coding it in multiple places. That applies to other hardcoded strings in the code. Something like:
https://github.com/smartdevicelink/sdl_java_suite/blob/797b7a14a7fd67fdf6faa504865e4e22228858ef/android/sdl_android/src/main/java/com/smartdevicelink/transport/SdlRouterService.java#L186